apps/studio/src/App.vue

```diff
@@ -42,194 +33,538 @@
-    ...mapState('licenses', {
+    ...mapState(["storeInitialized", "connected", "database"]),
+    ...mapState("licenses", {
       status: (state) => state.status,
       licensesInitialized: (state) => state.initialized,
     }),
     ...mapGetters({
-      'isTrial': 'isTrial',
-      'isUltimate': 'isUltimate',
-      'themeValue': 'settings/themeValue',
-    })
+      isTrial: "isTrial",
+      isUltimate: "isUltimate",
+      themeValue: "settings/themeValue",
+    }),
   },
   watch: {
     database() {
-      log.info('database changed', this.database)
+      log.info("database changed", this.database);
     },
-    themeValue() {
-      document.body.className = `theme-${this.themeValue}`
+    themeValue: {
+      handler(newTheme) {
+        console.log("Theme value changed in App.vue:", newTheme);
+
+        if (typeof newTheme === "string") {
+          // Handle string theme ID
+          this.applyThemeCSS(newTheme);
+        } else if (newTheme && newTheme.themeId) {
+          // Handle object with themeId and possibly CSS
+          this.applyThemeCSS(
+            newTheme.themeId,
+            newTheme.css,
+            newTheme.baseTheme
+          );
+        }
+      },
+      immediate: true,
     },
     status(curr, prev) {
-      this.$store.dispatch('updateWindowTitle')
-      this.validateLicenseExpiry(curr, prev)
+      this.$store.dispatch("updateWindowTitle");
+      this.validateLicenseExpiry(curr, prev);
     },
     async licensesInitialized(initialized) {
       if (initialized) {
-        await this.$nextTick()
-        this.validateLicenseExpiry()
+        await this.$nextTick();
+        this.validateLicenseExpiry();
       }
     },
   },
   async beforeDestroy() {
-    clearInterval(this.interval)
-    clearInterval(this.licenseInterval)
+    clearInterval(this.interval);
+    clearInterval(this.licenseInterval);
   },
   async mounted() {
-    this.notifyFreeTrial()
-    this.interval = setInterval(this.notifyFreeTrial, globals.trialNotificationInterval)
-    this.$store.dispatch('licenses/updateAll');
-    this.licenseInterval = setInterval(
-      () => this.$store.dispatch('licenses/updateAll'),
-      globals.licenseCheckInterval
-    )
-    const query = querystring.parse(window.location.search, { parseBooleans: true })
-    if (query) {
-      this.url = query.url || null
-      this.runningWayland = !!query.runningWayland
-    }
+    try {
+      this.notifyFreeTrial();
+      this.interval = setInterval(
+        this.notifyFreeTrial,
+        globals.trialNotificationInterval
+      );
+      this.$store.dispatch("licenses/updateAll");
+      this.licenseInterval = setInterval(
+        () => this.$store.dispatch("licenses/updateAll"),
+        globals.licenseCheckInterval
+      );
 
+      // Initialize settings
+      await this.$store.dispatch("settings/initializeSettings");
 
-    this.$nextTick(() => {
-      window.main.isReady();
```

Comment:
Can we move all of this theme logic into it's own component? Maybe <theme-manager>?

Reply:
👍🏼